### PR TITLE
Encryption: Add ingress encryption metrics and monitor output

### PR DESCRIPTION
**Pull request by jrfastab**
***IMPORTANT***
I use the upper bit of the reason field in the trace_notify message to indicate if a packet is encrypted or not. Any objects processing the trace notify event will need to be able to parse the bit, this can be as simple as masking it out. It is also only used when encryption is enabled users that do not enable encryption will not need to do anything. The change in datapath_trace.go is,
```
 r := reason & ^TraceReasonEncryptMask
```
***

Add metrics and trace output to ingress side of Encryption.

On the monitor add an "encrypt" flag to denote that the flow was decrypted. So now we see
  <- overflow ...
  <- overflow decrypted

Then on metrics side add a new ingress bytes/packets count to show decrypted packets,
```
REASON                       DIRECTION   PACKETS   BYTES
Interface                  Ingress     305164    61427600 
Interface Decrypted        Ingress     298241    44024299
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8048)

### CODE REVIEW

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
@tgraf any opinions on using a  bit of the reason field in the trace msg? Do we consider trace output stable API?

**Comment by jrfastab** [comment]
resolved merge conflict,  by updating bpf.sha will retest now.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
Failed connectivity test, retesting all that has changed is bpf.sha so should be unrelated,
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/12499/artifact/76e6cadd_K8sServicesTest_Checks_ClusterIP_Connectivity_Checks_service_on_same_node.zip

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

File: pkg/monitor/datapath_trace.go [hunk 1]
```diff
@@ -126,9 +140,9 @@ func (n *TraceNotify) traceSummary() string {
 
 // DumpInfo prints a summary of the trace messages.
 func (n *TraceNotify) DumpInfo(data []byte) {
-	fmt.Printf("%s flow %#x identity %d->%d state %s ifindex %s: %s\n",
-		n.traceSummary(), n.Hash, n.SrcLabel, n.DstLabel,
-		connState(n.Reason), ifname(int(n.Ifindex)), GetConnectionSummary(data[TraceNotifyLen:]))
+	fmt.Printf("%s %sflow %#x identity %d->%d state %s ifindex %s: %s\n",
```
Comment (aanm) [inline pkg/monitor/datapath_trace.go]: wouldn't this be `%s-flow` instead of `%sflow`?

**Review by aanm — APPROVED** [review]
All good, only a small nit.

File: pkg/monitor/datapath_trace.go (hunk 1 above)
Comment (jrfastab) [reply pkg/monitor/datapath_trace.go]: Yeah that will print nicer. Will push update.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
ah adding a '-' to monitor broke runtime tests using regex to pull out fields.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
fixed regex CI tests by changing monitor output from "encrypted-flow" to "encrypted flow" this way when encryption is disabled we don't have this "-flow" which broke regex tests over monitor output in CI.

**Comment by jrfastab** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Comment by jrfastab** [comment]
OK actually the regex is white space sensitive and I added an extra space in the case without encryption. I guess this is not desired so if/else case if the string exists or not and only print the encryption reason if its not the empty string. This way we don't have extra spaces on the line.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
The handful of errors above are due to CI regex not matching now that its resolved I'll merge this.

**Pull request merged**